ocaml/xapi/xapi_xenops.ml

```diff
@@ -1007,6 +1007,23 @@ end
 module Xenopsd_metadata = struct
 	(** Manage the lifetime of VM metadata pushed to xenopsd *)
 
+	(* If the VM has Xapi_globs.persist_xenopsd_md -> filename in its other_config,
+	   we persist the xenopsd metadata to a well-known location in the filesystem *)
+	let maybe_persist_md ~__context ~self md =
+		let oc = Db.VM.get_other_config ~__context ~self in
+		if List.mem_assoc Xapi_globs.persist_xenopsd_md oc then begin
+			let file_path =
+				Filename.concat Xapi_globs.persist_xenopsd_md_root (List.assoc Xapi_globs.persist_xenopsd_md oc) |>
+				Stdext.Unixext.resolve_dot_and_dotdot in
+
+			if not (String.startswith Xapi_globs.persist_xenopsd_md_root file_path) then begin
```

Comment:
Yes - this is to protect against someone saying "persist_xenopsd_md=../../../etc/passwd" or something similar.